File: src/main/java/com/DavidM1A2/othertest/common/worldGeneration/structure/base/processor/LowestHeightChunkProcessor.java
package com.DavidM1A2.afraidofthedark.common.worldGeneration.structure.base.processor;

import com.DavidM1A2.afraidofthedark.common.capabilities.world.IHeightmap;
import net.minecraft.util.math.ChunkPos;

/**
 * Utility processor for finding the minimum ground height within a region
 */
public class LowestHeightChunkProcessor implements IChunkProcessor<Integer>
{
    // The minimum ground height
    private int minGroundHeight = 256;
    // Heightmap to use
    private IHeightmap heightmap;

    /**
     * Constructor initializes the heightmap field
     *
     * @param heightmap The heightmap to use
     */
    public LowestHeightChunkProcessor(IHeightmap heightmap)
    {
        this.heightmap = heightmap;
    }

    /**
     * Processes the X,Z chunk by finding the lowest point
     *
     * @param chunkPos The coordinate of the chunk
     * @return true to continue processing
     */
    @Override
    public boolean processChunk(ChunkPos chunkPos)
    {
        // Compute the ground height in the chunk
        int groundHeight = heightmap.getLowestHeight(chunkPos);
        minGroundHeight = Math.min(minGroundHeight, groundHeight);
        return true;
    }

    /**
     * @return The minimum ground height
     */
    @Override
    public Integer getResult()
    {
        return minGroundHeight;
    }
}

File: src/main/java/com/davidm1a2/afraidofthedark/common/worldGeneration/structure/base/processor/LowestHeightChunkProcessor.kt
package com.davidm1a2.afraidofthedark.common.worldGeneration.structure.base.processor

import com.davidm1a2.afraidofthedark.common.capabilities.world.IHeightmap
import net.minecraft.util.math.ChunkPos
import kotlin.math.min

/**
 * Utility processor for finding the minimum ground height within a region
 *
 * @constructor initializes the heightmap field
 * @param heightmap The heightmap to use
 * @property minGroundHeight The minimum ground height
 */
class LowestHeightChunkProcessor(private val heightmap: IHeightmap) : IChunkProcessor<Int> {
    private var minGroundHeight = 256

    /**
     * Processes the X,Z chunk by finding the lowest point
     *
     * @param chunkPos The coordinate of the chunk
     * @return true to continue processing
     */
    override fun processChunk(chunkPos: ChunkPos): Boolean {
        // Compute the ground height in the chunk
        val groundHeight = heightmap.getLowestHeight(chunkPos)
        minGroundHeight = min(minGroundHeight, groundHeight)
        return true
    }

    /**
     * @return The minimum ground height
     */
    override fun getResult(): Int {
        return minGroundHeight
    }
}